src/observableApiConfig.ts

```diff
@@ -68,9 +68,11 @@ export async function setObservableApiKey(info: null | {id: string; key: string}
 
 export async function getDeployConfig(
   sourceRoot: string,
+  requestedDeployConfigPath: string | null = null,
   effects: ConfigEffects = defaultEffects
 ): Promise<DeployConfig> {
-  const deployConfigPath = op.join(effects.cwd(), sourceRoot, ".observablehq", "deploy.json");
+  const deployConfigPath =
+    requestedDeployConfigPath ?? op.join(effects.cwd(), sourceRoot, ".observablehq", "deploy.json");
```

Comment:
I have a slight preference for only applying the default when `requestedDeployConfigPath` is undefined, rather than treating null and undefined as the same. A reasonable alternative interpretation of the null config path is that we want to force the deploy not use a config (neither read nor write); we don’t support that now but perhaps we might want to reserve it for future use.

Also, the `requestedDeployConfigPath` feels a bit verbose ([enterprise-y](https://projects.haykranen.nl/java/)). I think it’d be reasonable to just call it `deployConfigPath` and have a default value assigned in-place rather than having separate variables.